Final Project/Server/fiimentor/controllers/authController.ts

```diff
@@ -1,29 +1,89 @@
 import HttpStatus from "http-status-codes";
-
+import {createConnection} from 'typeorm';
+import {UserRepository} from '../Repositories/UserRepository'
 import {createToken} from '../utils';
-
+import {ProfessorRepository} from "../Repositories/ProfessorRepository";
+import {StudentRepository} from "../Repositories/StudentRepository";
+import {TutorRepository} from "../Repositories/Tutor";
+import {error} from "util";
+import {User} from "../models/entities/User";
 async function login(req: any, res: any){
     try {
-        // Mock user
-        const user = {
-            id: 1,
-            username: 'teodor',
-            email: 'teodor@gmail.com'
-        };
-
         const username = req.body.username;
         const password = req.body.password;
+        let student : any =null;
+        let professor : any =null;
+        let tutor : any =null;
+        createConnection().then(async connection => {
+            const userRepository = new UserRepository(connection);
+            const userByUserName = await userRepository.getByUsername(username);
 
-        //serach in the database if user is existent
-
+            if (userByUserName.length)
+            {
+                const role=userByUserName[0].role;
+                if (role.localeCompare("student")==0) {
+                    const studentRepository = new StudentRepository(connection);
+                    student = await studentRepository.getByUserId(userByUserName[0].id);
+                }
+                if (role.localeCompare("professor")==0){
+                    const professorRepository = new ProfessorRepository(connection);
+                    const tutorRepository = new TutorRepository(connection);
+                    professor=await professorRepository.getByUserId(userByUserName[0].id);
+                    tutor=await tutorRepository.getByProfessorId(professor[0].id);
+                }
+            }
 
-        // Creating a token to return to the client
-        const token = await createToken(user, process.env.JWT_SECRET);
-        //send the token to the client
+            await connection.close();
+            if (!userByUserName.length){
+                return res.status(HttpStatus.UNAUTHORIZED).json({
+                    success: false,
+                    status: 'Invalid username'
+                });
+            }
+            else{
+                if(password.localeCompare(userByUserName[0].password)!=0){
+                    return res.status(HttpStatus.UNAUTHORIZED).json({
+                        success: false,
+                        status: 'Invalid password'
+                    });
+                }
```

Comment:
cred ca era mai ok sa faci verificarea toata pt user cu tot cu password inainte de etapa de mai sus cand iti aduci din bd student sau prof in functie de rol